sources/packages/web/src/views/student/StudentApplicationSummary.vue

```diff
@@ -38,14 +57,86 @@ export default {
     StudentApplications,
     CheckValidSINBanner,
     HeaderNavigator,
+    ConfirmEditApplication,
+    CancelApplication,
   },
   setup() {
     const hasRestriction = ref(false);
     const restrictionMessage = ref("");
+    const programYear = ref({} as ProgramYearOfApplicationDto);
+    const router = useRouter();
+    const toast = useToastMessage();
+    const TOAST_ERROR_DISPLAY_TIME = 15000;
+    const editApplicationModal = ref({} as ModalDialog<boolean>);
+    const showModal = ref(false);
+    const selectedApplicationId = ref(0);
+    const reloadData = ref(false);
+
+    const openConfirmCancel = (id: number) => {
+      showModal.value = true;
+      selectedApplicationId.value = id;
+    };
+
+    const showHideCancelApplication = () => {
+      showModal.value = !showModal.value;
+    };
+
+    const getProgramYear = async (applicationId: number) => {
+      programYear.value = await ApplicationService.shared.getProgramYearOfApplication(
+        applicationId,
+      );
+    };
+
+    const editApplication = async (applicationId: number) => {
+      try {
+        await getProgramYear(applicationId);
+        router.push({
+          name: StudentRoutesConst.DYNAMIC_FINANCIAL_APP_FORM,
+          params: {
+            selectedForm: programYear.value.formName,
+            programYearId: programYear.value.programYearId,
+            id: applicationId,
+          },
+        });
+      } catch (error) {
+        toast.error(
+          "Program Year not active",
+          undefined,
+          TOAST_ERROR_DISPLAY_TIME,
+        );
+      }
+    };
+
+    const confirmEditApplication = async (applicationId: number) => {
+      if (await editApplicationModal.value.showModal()) {
+        editApplication(applicationId);
+      }
+    };
+
+    const editApplicationAction = async (
+      status: ApplicationStatus,
+      applicationId: number,
+    ) => {
+      if (status !== ApplicationStatus.draft)
+        confirmEditApplication(applicationId);
+      else editApplication(applicationId);
+    };
+
+    const setReloadData = () => {
+      reloadData.value = true;
+    };
+
+    const goToApplication = (id: number) => {
+      return router.push({
+        name: StudentRoutesConst.STUDENT_APPLICATION_DETAILS,
+        params: {
+          id: id,
+        },
+      });
+    };
```

Comment:
router.push is a promise. Please refer to my previous comment on using async/await.